### ISSUE TITLE
unsafe fixed buffers. nameof doesn't work

### ISSUE THREAD

**Issue by mikasoukhov** [dotnet/roslyn#10467]
6.0 (from vs 2015): 

```
unsafe struct Struct1
{
  public fixed char MessageType[50];

  public override string ToString()
  {
    return nameof(MessageType) + "=Struct1";// CS1666
  }
}
```

**Comment by jaredpar** [comment]
Thanks for reporting this issue.  We will take a look at this.

**Comment by TyOverby** [comment]
Here's another failing example that @brettfo came up with:

```cs
    void foo(ref int i)
    {
        Action a = () =>
        {
            var x = nameof(i); // Can not use ref or out parameter `i` inside of anonymous method, lambda, ...
        };
    }
```

The common pattern is reporting these errors before we know if nameof is a function, or a keyword.

Interestingly, this reports no errors:

```cs
class O
{
    int i;
    static void foo()
    {
        var x = nameof(i);
    }
}
```

I think the solution is to delay reporting these errors until we have a semantic model and can tell if nameof is a method or not.
